### ISSUE TITLE
Add rules for leaf_cycle=* and leaf_type=*

### ISSUE THREAD

**Issue by BushmanK** [gravitystorm/openstreetmap-carto#1756]
Once rendering style for natural=wood has changed recently, it seems like it's time to think about recently introduced, but quite popular tags such as **leaf_cycle** and **leaf_type** (more that 1 million entries in the whole world).

I made several sample patterns using current pattern used for natural=wood.

leaf_type=needleleaved no leaf_cycle
![needleleaved](https://habrastorage.org/files/bc6/fba/7c5/bc6fba7c5bcf4f1eb684c4cafc103ddf.png)

leaf_type=broadleaved no leaf_cycle
![broadleaved](https://habrastorage.org/files/4a9/241/392/4a9241392ed14f18bb947cb831f8b57c.png)

leaf_type=broadleaved leaf_cycle=deciduous
![broadleaved deciduous](https://habrastorage.org/files/6e4/2aa/918/6e42aa918bed495fb2eac710cd720a0b.png)

For leaf_cycle=evergreen, leaf_cycle=semi_evergreen,  leaf_cycle=semi_deciduous I suggest using pattern with fully filled tree outlines, but any other ideas are welcome, for sure.

So, basically, for unknown life cycle outline should be empty; for trees, looking differently through the year - half-filled, for trees, not changing or barely changing its appearance - fully filled.

While not all needle-leaved plants have similar conic shape as fir or cypress (mature pine, for example), in majority of cases, this shape is associated with them pretty well. So, I believe, using conic and rounded canopy outline is legitimate way to show broadleaved and needleleaved trees.

Actually, I have no  idea how to show leafless plants - hard to imagine unified symbol for them, they are too different.

**Comment by matkoniecz** [comment]
leaf_type is covered by #822

**Comment by BushmanK** [comment]
I see.
It seems like my idea of empty, half-filled and filled outlines is new.

**Comment by pnorman** [comment]
Closing in favour of #822, as there is substantial overlap.

**Comment by kocio-pl** [comment]
@BushmanK Nice new idea, please tell about it on #822.

**Comment by kocio-pl** [comment]
Reopening, because leaf_cycle=* is still not implemented.

Simple try with filling half (semi_/deciduous) or full (semi_/evergreen) areas:

![broadleaved-deciduous](https://user-images.githubusercontent.com/5439713/30044062-9391dcc8-91fb-11e7-830a-3b4548886818.png)
![broadleaved-evergreen](https://user-images.githubusercontent.com/5439713/30044064-9706484e-91fb-11e7-807c-dc6114806f6c.png)
![needleleaved-deciduous](https://user-images.githubusercontent.com/5439713/30044065-99c50fa2-91fb-11e7-8cf7-995b4c24b6f2.png)
![needleleaved-evergreen](https://user-images.githubusercontent.com/5439713/30044072-9c398c5e-91fb-11e7-9474-fffd1886fc7a.png)
![mixed-deciduous](https://user-images.githubusercontent.com/5439713/30044074-9e252bcc-91fb-11e7-8cd7-a26dffe0d779.png)
![mixed-evergreen](https://user-images.githubusercontent.com/5439713/30044078-a01f1122-91fb-11e7-8e22-01d9ab00f7dd.png)

**Comment by pnorman** [comment]
I think this is trying to cram too much non-obvious detail into the symbology

**Comment by math1985** [comment]
+1

**Comment by kocio-pl** [comment]
Do you think this is just suboptimal solution or leaf_cycle is too much to fit in general style and we should close the issue as wontfix?

**Comment by matkoniecz** [comment]
Yeah, seconds after reading description at https://github.com/gravitystorm/openstreetmap-carto/issues/1756#issuecomment-327063068 I am forgetting meaning of symbols.

I see no reasonable way to show that.

**Comment by imagico** [comment]
Keep in mind that rendering leaf_type but not leaf_cycle would bring in a significant amount of additional geographical bias - the almost completely binary distinction of deciduous broadleaved and evergreen needleleaved is something relatively specific to the northern hemisphere temperate zone you all live in.  In other parts of earth leaf_cycle is often a much more significant distinction than leaf_type.

**Comment by kocio-pl** [comment]
> Yeah, seconds after reading description at #1756 (comment) I am forgetting meaning of symbols.

For me it's easy:
- **full** filling - leaves **always** present
- **partial** filling - leaves **not always** present
- no filling - unknown leaf cycle (as currently rendered, so I didn't repeat it)

It just makes a 9 combinations matrix with leaf types, so it looks like a lot of symbols, but it's just 3x3 (plus leafless as a special case).

**Comment by pnorman** [comment]
> Do you think this is just suboptimal solution or leaf_cycle is too much to fit in general style and we should close the issue as wontfix?

wontfix, I just don't see it as possible.

> It just makes a 9 combinations matrix with leaf types, so it looks like a lot of symbols, but it's just 3x3 (plus leafless as a special case).

It's better than 9 distinct symbols, but it's still too much. I have other concerns with the details of symbols, but the main one is that it's too much for a general style.

**Comment by kocio-pl** [comment]
I guess we should close it now for practical reasons (I see open tickets as worth fixing and this one doesn't look that way). If a more acceptable solutions will ever show up, it can be reopened of course.

**Comment by imagico** [comment]
Reopening this as far as `leaf_cycle` is concerned because:

* it was closed for unclear reason (not worth fixing or not feasible fixing).
* plenty of options to illustrate leaf_cycle in a pattern are known and some have been demonstrated on digital maps - see for example [here](https://imagico.de/blog/en/differentiated-rendering-of-woodland-in-maps/).
* as pointed out above in https://github.com/gravitystorm/openstreetmap-carto/issues/1756#issuecomment-327117703 preferentially distinguishing `leaf_type` rather than `leaf_cycle` is communicating a significant geographic bias because both the correlation between `leaf_type` and `leaf_cycle` and the higher significance of `leaf_type` in practical differentiation of woodlands is quite specific to the northern hemisphere temperate zone.
* use of `leaf_cycle` on woodland polygons has grown a lot since this was closed, it numbers nearly 1 million now - which is more than `leaf_type` back in 2017.  Also percentage of woodland polygons tagged with `leaf_cycle` has increased quite a lot:

tag | 2018 | with `leaf_cycle` | percent | 2022 | with `leaf_cycle` | percent
--- | --- | --- | --- | --- | --- | ---
`natural=wood` | 4.6M | 111k | **2.4** | 8.5M | 634k | **7.5**
`landuse=forest` | 3.5M | 115k | **3.3** | 5.0M | 333k | **6.7**
